### ISSUE TITLE
Can this be de-obfuscated?

### ISSUE THREAD

**Issue by fritx** [mikrofusion/gulp-obfuscate#8]
This is awesome, but I haven't used it yet.
Btw, Is there any technology to anti-obfuscate this?
I wonder is it okay to be used in production environment.
Cheers.

**Comment by mikrofusion** [comment]
Thanks @fritx.  

This obfuscates the variables in your code in a destructive manner - i.e. ``` var foo; and var bar; ``` will become something like ``` var a1; var a2; ```.  Once the variables have been changed it shouldn't be possible to convert them back without some key pair table which the user would not have (i.e. foo => a1, bar => a2).  So they should never be able to get back to the original source code.

That said, this only obfuscates variables.  If someone really wanted to take the time to figure out some algorithm you used then they still could; In reality its not really possible to completely hide client side algorithms because as long as the browser can interpret the code, then so can a user.  The goal for this is to make it so difficult to read / understand that it wouldn't be worth their time.

As far as running in production, it should be fine - but as with anything that causes changes to code I would make sure I ran it through QA (in the obfuscated mode) before releasing to ensure there are no issues.
